backend/main/src/Group/Query/FindContainersInAGroupQuery/FindContainersInAGroupQuery.ts

```diff
@@ -0,0 +1,91 @@
+import { IQuery } from "src/Shared/Layers/Query/IQuery";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import { isValidUUIDV4 } from "src/Shared/Utils/Validator";
+import { ContainerData } from "src/Shared/Adapters/DB/NishikiDBTypes";
+import { Ok, Err, Result } from "result-ts-type";
+import { QueryError } from "src/Shared/Utils/Errors";
+
+export class FindContainersInAGroupQuery
+	implements
+		IQuery<{ groupId: string }, IContainerData, InvalidUUId | GroupNotFound>
+{
+	constructor(private readonly nishikiDynamoDBClient: NishikiDynamoDBClient) {}
+
+	public async execute(input: { groupId: string }): Promise<
+		Result<IContainerData, InvalidUUId | GroupNotFound>
+	> {
+		const { groupId } = input;
+
+		if (!isValidUUIDV4(groupId)) {
+			return Err(new InvalidUUId("invalid group ID is provided"));
+		}
+
+		const [containerIds, groupData] = await Promise.all([
+			this.nishikiDynamoDBClient.listOfContainers(groupId),
+			this.nishikiDynamoDBClient.getGroup({ groupId }),
+		]);
+
+		if (!groupData) {
+			return Err(new GroupNotFound("Requested group is not found"));
+		}
+
+		if (containerIds.length === 0) {
+			return Ok({ containers: [] });
+		}
+
+		const containersResult = await Promise.all(
+			containerIds.map((containerId) =>
+				this.nishikiDynamoDBClient.getContainer(containerId),
+			),
+		);
+
+		const containers = containersResult.filter(
+			(container) => container,
+		) as ContainerData[];
+
+		return Ok({
+			containers: containers.map((container) => ({
+				id: container.containerId,
+				name: container.containerName,
+				group: {
+					groupId: groupData.groupId as string,
+					groupName: groupData.groupName as string,
```

Comment:
I think these two values can't be anything other than the String type. Is there any reason that you are using `as string` here?

Reply:
Thank you for pointing it out.
I added when implementing this module, and forgot to delete it.
Now I removed it.